pkg/controllers/netpol/utils.go

```diff
@@ -142,3 +143,84 @@ func getPodIPv4Address(pod podInfo) (string, error) {
 	}
 	return "", fmt.Errorf("pod %s has no IPv4Address", pod.name)
 }
+
+// safeJoin joins the namespace and name, ensuring that the result is less than or equal to 48 characters
+func safeJoin(namespace string, name string) string {
+	if (len(namespace) + len(name)) < 48 {
+		return namespace + "/" + name
+	}
+
+	// We must create at least one substring
+	if len(namespace) < 24 {
+		lengthSubString := (48 - len(namespace) - 1)
+		return namespace + "/" + name[0:lengthSubString]
+	}
+
+	if len(name) < 24 {
+		lengthSubString := (48 - len(name) - 1)
+		return namespace[0:lengthSubString] + "/" + name
+	}
+
+	//If we arrive here, both are over 24 characters
+	return namespace[0:23] + "/" + name[0:23]
+}
+
+// getIptablesNFlogLimit reads the annotations setting the nflog limit and limit-burst config
+// "kube-router.io/netpol-nflog-limit" and "kube-router.io/netpol-nflog-limit-burst"
+func getIptablesNFlogLimit(annotations map[string]string) (string, string, error) {
+	defaultLimit := "10/minute"
+	defaultLimitBurst := "10"
+
+	limit, ok := annotations["kube-router.io/netpol-nflog-limit"]
+	if !ok {
+		limit = defaultLimit
+	}
+
+	limitBurst, ok := annotations["kube-router.io/netpol-nflog-limit-burst"]
+	if !ok {
+		limitBurst = defaultLimitBurst
+	}
+
+	if err := checkNFlogParams(limit, limitBurst); err != nil {
+		return "", "", err
+	}
+
+	return limit, limitBurst, nil
+}
+
+
+// checkNFlogParams verifies that the nflog parameters are correct
+// * kube-router.io/netpol-nflog-limit must be an integer, with an optional "/second", "/minute", "/hour", "/day" or the first character of each time unit
+// * kube-router.io/netpol-nflog-limit-burst must be an integer
+func checkNFlogParams(limit string, limitBurst string) error {
+	stringSuffixes := []string{"second", "minute", "hour", "day", "s", "m", "h", "d"}
+
+	// If limit does not set a time unit, check if limit and limitBurst are integers 
+	param := strings.Split(limit, "/")
+	if len(param) == 1 {
+		if isInteger(param[0]) && isInteger(limitBurst) {
+			return nil
+		}
+	}
+	
+	// If limit sets a time unit, check if it is among the supported ones
+	if len(param) == 2 {
+		for _,suffix := range(stringSuffixes) {
```

Comment:
If we use `sets.String` this could just be `stringSuffixes.Has(param[1])` instead of iterating over the whole list every time.